tango_simlib/tests/test_fandango_json_parser.py

```diff
@@ -0,0 +1,196 @@
+        'delta_val': 'Not specified',
+        'extensions': '[]',
+        'max_alarm': 'Not specified',
+        'max_warning': 'Not specified',
+        'min_alarm': 'Not specified',
+        'min_warning': 'Not specified'
+        },
+    'color': 'Lime',
+    'data_format': tango._tango.AttrDataFormat.SCALAR,
+    'data_type': tango._tango.CmdArgType.DevString,
+    'database': 'monctl:10000',
+    'description': '',
+    'device': 'sys/database/2',
+    'display_unit': 'No display unit',
+    'events': {
+        'arch_event': {
+            'archive_abs_change': 'Not specified',
+            'archive_period': 'Not specified',
+            'archive_rel_change': 'Not specified',
+            'extensions': '[]'
+            },
+        'ch_event': {
+            'abs_change': 'Not specified',
+            'extensions': '[]',
+            'rel_change': 'Not specified'
+            },
+        'per_event': {
+            'extensions': '[]',
+            'period': '1000'
+            }
+        },
+    'format': '%s',
+    'label': 'Status',
+    'max_alarm': 'Not specified',
+    'min_alarm': 'Not specified',
+    'model': 'monctl:10000/sys/database/2/Status',
+    'name': 'Status',
+    'polling': 0,
+    'quality': 'ATTR_VALID',
+    'standard_unit': 'No standard unit',
+    'string': 'Device is OK',
+    'time': 1521724934.799581,
+    'unit': '',
+    'value': 'Device is OK',
+    'writable': 'READ'
+    }
+
+
+class GenericSetup(unittest.TestCase):
+    """A class providing the setUp method definition for the other test classes.
+    """
+    longMessage = True
+
+    def setUp(self):
+        super(GenericSetup, self).setUp()
+        self.fandango_json_file = pkg_resources.resource_filename(
+            'tango_simlib.tests.config_files', 'database2.json')
+        self.fandango_parser = fandango_json_parser.FandangoExportDeviceParser()
+        self.fandango_parser.parse(self.fandango_json_file)
+
+
+class TestFandangoJsonParser(GenericSetup):
+    """A test class that tests that the fandango json parser works correctly.
+    """
+    def test_preprocess_command_types(self):
+        """Testing that command property mapping was done well and command type
+        of mapped properties is of tango type.
+        """
+
+        # testing that the command properties have been renamed to
+        # match values in command signature
+        for cmd_name in self.fandango_parser.get_device_command_metadata():
+            self.assertEquals(
+                EXPECTED_CMD_PROPERTY_PARAMETERS.sort(),
+                self.fandango_parser.get_device_command_metadata()
+                [cmd_name].keys().sort(),
+                "The mapping of the command properties to the command"
+                "signature wasn't successful for '%s' command" % (cmd_name)
+                )
+            # testing that data type of cmd property values is a tango type and
+            # there is no occurence of 'Const' in the value of the cmd property
```

Comment:
https://github.com/ska-sa/tango-simlib/blob/741fc93424acdce730e3fab6c37a748f31447207/tango_simlib/utilities/fandango_json_parser.py#L68-L69
Yes, that's right. It's not in the json file but I wrote the test for it because of the lines above in the fandango_json_parser script

Reply:
It looks like it is a leftover from the XMI parser:
https://github.com/ska-sa/tango-simlib/blob/533e605/tango_simlib/utilities/sim_xmi_parser.py#L413-L416

I don't think we need it in the fandango parser, and then it can be removed from the test as well.
@kmadisa Do you agree?

Reply:
Oh I see. I'll effect the change.

Reply:
@ajoubertza Yes it  is not needed here.